src/main/java/com/adyen/httpclient/HttpURLConnectionClient.java

```diff
@@ -38,6 +38,7 @@
 public class HttpURLConnectionClient implements ClientInterface {
     private static final String CHARSET = "UTF-8";
     private Proxy proxy;
+    private int connectTimeout = -1;
```

Comment:
We can give it a more meaningful name for example:  connectionTimeoutMillis

Reply:
Naming agreed. Could you explain a little bit more what you mean with Config and Environment?
Shall I add a field for timeout in Config as ell and check it in the HttpURLConnectionClient.java ?
If yes, which one should have priority?

Reply:
Yes. A new field will be added in Config.java.
You may check an example of how Configuration setup for (x-api-key) is done here:
https://adyen.github.io/adyen-java-api-library/index.html

Usually, Config object is used for setting the environment/configuration settings (As shown in example) and HttpURLConnectionClient is not directly created so only Config object is setting connctionTimeOut.

Since you are following proxy settings section, hence you are creating a client directly.

So, in short, either Config or HttpURLConnectionClient will be used for setting timeout so at a time only one value will be there.